### ISSUE TITLE
UI: terminal tabs

### ISSUE THREAD

**Issue by christianparpart** [contour-terminal/contour#90]
Provide terminal tabs.

Depends on #956 

This requires basic support for the tree structure windowing.

* per output window, only one session is being displayed (no session management in this ticket yet).
* one session can contain multiple tabs (by default being displayed in the bottom line)
* one tab can contain multiple panes (tree structured)
* one pane contains exactly one single terminal view

## Contour: Actions

* [ ] NewTab (Ctrl+Shift+T)
* [ ] SelectNextTab (Ctrl+Shift+PgDn)
* [ ] SelectPrevTab (Ctrl+Shift+PgUp)
* [ ] SelectTab(index) (Ctrl+Alt+<index>)
* [ ] CloseTab (Ctrl+Shift, C)

## TODO  Items

* [ ] TerminalTree: terminal tree data structure (session/panel/pane)
* [ ] action: rename session (default: indexed)
* [ ] replace single-TerminalView with TerminalTree
* [ ] action: rename panel (default: indexed)
* [ ] action: rename pane (default: terminal title)
* [ ] action bindings

**Comment by poige** [comment]
> CloseTab (Ctrl+Shift, C)

On Linux it'd be a conflicting habit with some of the existing terminals that use this to Copy.
Also, from my own experience I'd suggest implementing MRU tab switching, it's certainly a "killer" feature that iTerm has and lots of terminals lack of.

**Comment by christianparpart** [comment]
These where initial shortcut ideas m but yes, that will ask be certainly configurable as usual.

Many thanks on the iTerm tip @poige , also. :-)

**Comment by denisgolius** [comment]
Any updates on this?

PS: I also interested in feature to split terminal window like [Tilix](https://github.com/gnunn1/tilix) does
![image](https://github.com/contour-terminal/contour/assets/5650611/c0858970-517a-4fe0-be19-e6cec7ab18ca)
Yes, I saw https://github.com/contour-terminal/contour/issues/91 but there's been no change since the creation of these two

**Comment by christianparpart** [comment]
Hey @denisgolius. Many thanks for your continued interest in Contour :)

This is still pretty high on the list. We're currently about to release 0.4.0, which contains the major frontend refactor, which is considered a prerequisite for also this ticket here.

As soon as we've release 0.4.0 (which requires #1228 to be fixed), I can start working on tabs and then splits. :)

**Comment by binex-dsk** [comment]
Why not basic tab support in one release, and more complex pane and tree structures in a later release? Having even basic tabbing support would be a huge step up for user friendliness.

**Comment by christianparpart** [comment]
Good morning. This is a good idea. We will however first release 0.4.0 ASAP, and then care about Tabs. Splits need more thought, but can be easily deferred, you are right. :)
